pkg/logs/logs.go

```diff
@@ -58,19 +61,45 @@ func (writer GlogWriter) Write(data []byte) (n int, err error) {
 }
 
 // InitLogs initializes logs the way we want for kubernetes.
-func InitLogs(fs *flag.FlagSet) {
+func InitLogs() {
 	logs.InitLogs()
 
-	if fs == nil {
-		fs = flag.CommandLine
-	}
-	initDeprecatedFlags(fs)
-	_ = fs.Set("logtostderr", "true")
-
 	log.SetOutput(GlogWriter{})
 	log.SetFlags(0)
 }
 
+func AddFlags(opts *logs.Options, fs *pflag.FlagSet) {
+	// init deprecated flags
+	{
+		var allFlags flag.FlagSet
+		klog.InitFlags(&allFlags)
+
+		allFlags.VisitAll(func(f *flag.Flag) {
+			switch f.Name {
+			case "add_dir_header", "alsologtostderr", "log_backtrace_at", "log_dir", "log_file", "log_file_max_size",
+				"logtostderr", "one_output", "skip_headers", "skip_log_headers", "stderrthreshold":
+				fs.AddGoFlag(f)
+			}
+		})
+	}
+
+	{
```

Comment:
**nitpick:** These indented code blocks aren't necessary IMO. Plus, it breaks [Go's Unindented Line of Sight](https://maelvls.dev/go-happy-line-of-sight/) for no good reason. But that's a nitpick, feel free to ignore